apps/web/src/components/Profile/NftGallery/Create.tsx

```diff
@@ -16,40 +22,123 @@ interface Props {
 
 const Create: FC<Props> = ({ showModal, setShowModal }) => {
   const [emoji, setEmoji] = useState('');
-  const [pickNfts, setPickNfts] = useState(false);
+  const [currentStep, setCurrentStep] = useState('NAME');
+  const gallery = useNftGalleryStore((state) => state.gallery);
+  const setGallery = useNftGalleryStore((state) => state.setGallery);
+  const currentProfile = useAppStore((state) => state.currentProfile);
+
+  const [createGallery, { loading }] = useCreateNftGalleryMutation();
+
+  const closeModal = () => {
+    setShowModal(false);
+    setCurrentStep('NAME');
+    setGallery({ name: '', items: [] });
+  };
+
+  const create = async () => {
+    try {
+      const sanitizedItems = gallery.items.map((el) => {
+        return { tokenId: el.tokenId, contractAddress: el.contractAddress, chainId: el.chainId };
+      });
+      const result = await createGallery({
+        variables: {
+          request: {
+            items: sanitizedItems,
+            name: gallery.name,
+            profileId: currentProfile?.id
+          }
+        }
+      });
+      if (result) {
+        closeModal();
+        toast.success('Gallery created');
+      }
+    } catch {}
+  };
+
+  const validateInputs = () => {
+    const galleryName = trimify(gallery.name);
+    if (!galleryName.length) {
+      toast.error('Gallery name required');
+    } else if (!gallery.items.length && currentStep === 'PICK_NFTS') {
+      toast.error('Select collectibles for your gallery');
+    }
+  };
+
+  const onClickNext = async () => {
+    validateInputs();
+    const galleryName = trimify(gallery.name);
+    if (galleryName.length && gallery.items.length && currentStep === 'REVIEW') {
+      create();
+    } else if (galleryName.length && gallery.items.length) {
+      setCurrentStep('REVIEW');
+    } else if (galleryName.length && !gallery.items.length) {
+      setCurrentStep('PICK_NFTS');
+    } else {
+      setCurrentStep('NAME');
+    }
+  };
+
+  const onPickEmoji = (emoji: string) => {
+    setEmoji(emoji);
+    setGallery({ ...gallery, name: gallery.name + emoji });
+  };
+
+  const getModalTitle = () => {
+    if (currentStep === 'PICK_NFTS' || currentStep === 'REVIEW') {
+      return (
+        <div className="flex items-center space-x-1">
+          <button
+            type="button"
+            onClick={() =>
+              setCurrentStep(
+                currentStep === 'REVIEW' ? 'PICK_NFTS' : currentStep === 'PICK_NFTS' ? 'NAME' : 'NAME'
+              )
+            }
+          >
+            <ChevronLeftIcon className="h-4 w-4" />
+          </button>
+          <span>
+            <Trans>
+              {currentStep === 'REVIEW' ? 'Review collection' : 'Select collectibles you want others to see'}
+            </Trans>
+          </span>
+        </div>
+      );
+    }
+    return "What's your gallery name?";
```

Comment:
```suggestion
    }

    return "What's your gallery name?";
```

also wrap in `t` macro for i18n